libcaf_core/caf/binary_deserializer.hpp

```diff
@@ -33,17 +34,36 @@ class binary_deserializer final : public deserializer {
 
   using super = deserializer;
 
-  using buffer = std::vector<char>;
-
   // -- constructors, destructors, and assignment operators --------------------
 
-  binary_deserializer(actor_system& sys, const char* buf, size_t buf_size);
+  template <class ValueType>
+  binary_deserializer(actor_system& sys, const ValueType* buf, size_t buf_size)
```

Comment:
This signature can lead to very subtle bugs, because it has an unclear semantics now. If `ValueType` is anything with a size > 1 Byte, what is `buf_size` now meaning? Number of bytes of number of elements?